compiler/CCT/success/sct.jazz

```diff
@@ -0,0 +1,21 @@
+#[sct="{ ptr: transient, val : { n: d, s: secret } } -> d"]
+export
+fn load(reg ptr u32[1] a) -> reg u32 {
+  _ = #init_msf();
+  reg u32 r;
+  r = a[0];
+  return r;
+}
+
+#[sct="{n: a, s: secret} × {n: b, s: secret} → c"
```

Comment:
Is it a bug in the CT checker? Or, to put it differently, should the CT checker accept free level variables in return type annotations?

Reply:
My idea, is that we should be able to have a better translation from
constraint to max. 
a -> b -> b, a <= b              ---> a -> b -> max (a, b)
a -> b -> c, a <= c, b <= c  ---> a -> b -> max (a, b)
Since I am not sur that we can always do this translation, and obtain 
a type that is valid for CT, we can keep it like this for the moment. 
It is just that the actual translation is simply more partial that what I have in mind.

Reply:
I had understood that you would prefer doing the translation the other way round: use “max” in annotations and translate that to constraints for use in the actual checker.